README.md

```diff
@@ -0,0 +1,86 @@
+# Fewlines Readline
+
+This package uses the [NodeJs Readline](https://nodejs.org/api/readline.html) and wrap it in a custom package in order to simplify its use and testing with jest mocks in Typescript.
+
+## Usage
+
+`fwl-readline` is not yet available on npm.
+To install it, add this in your `package.json` file:
+
+### Installation
+
+```json
+"dependencies": {
+  "fwl-readline": "fewlines-education/fwl-readline#master"
+}
+```
+
+### Import
+```js
+// example.js
+import * as readline from "fwl-readline";
+
+```
+
+### Create a reader
+
+We decided to simplify the initial API of the readline package. For this reason, we only export the `createInterface` function.
+
+This functions takes an `object` parameter with to keys:
+- `input`: generally the `process.stdin`
+- `output`: generally the `process.stdout`
+
+```js
+export const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+```
+
+### Type
+
+In the example above, the created `rl` will have the `Reader` type:
+
+```ts
+interface Reader {
+  question: (query: string, callback: (answer: string) => void) => void;
+  close: () => void;
+}
+```
+
+### Use the created reader
+
+```js
+rl.question("Enter something\n> ", (input) => {
+  console.log(`User input: "${input}"`);
+  rl.close();
+});
+```
+
+This example will output:
+```bash
+Enter something
+> something
+User input: "something"
+```
+
+### Mock with jest
+
+```js
+jest.mock("fwl-readline", () => {
+  return {
+    createInterface: () => {
+      return {
+        question: () => {},
+        close: () => {},
+      };
+    },
+  };
+});
+
+import functionThatUseFwlRealine from "wherever-you-want"
+
```

Comment:
I feel like this part's explanation is a bit light for mocks but it would take a lot to properly re-explain function mocking with jest..

Reply:
In fact, i decided to not further explain it because it is highly dependant of what you want to test:
- a complete interface using readline?
- functions using readline?

If we want to have some examples we could still discuss it in team and even if someone asks for more details, I will answer point by point.